pkg/manager/internal.go

```diff
@@ -252,6 +263,57 @@ func (cm *controllerManager) serveMetrics(stop <-chan struct{}) {
 	}
 }
 
+func (cm *controllerManager) serveHealthProbes(stop <-chan struct{}) {
+	mux := http.NewServeMux()
+
+	if cm.readinessEndpointName != "" {
+		var readinessHandler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
+			if cm.isReady {
+				w.WriteHeader(200)
+				_, err := w.Write([]byte("ok\n"))
+				log.Error(err, "Failed to send readiness probe response")
+			} else {
+				w.WriteHeader(503)
+				_, err := w.Write([]byte("error\n"))
+				log.Error(err, "Failed to send readiness probe response")
+			}
+		}
+		mux.Handle(cm.readinessEndpointName, readinessHandler)
+	}
+	if cm.livenessEndpointName != "" {
+		var livenessHandler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
```

Comment:
we should copy the `k8s.io/apiserver` healthz code, and then switch to that whenever it moves to component-base (@luxas ?).  It provides support for pluggable health checks, so we could have multiple in a single manager (e.g. health check per controller).

Reply:
Why can't we directly use `k8s.io/apiserver`? Its description is `Library for writing a Kubernetes-style API server.` If it's a lib, why can't we just use it instead of copying it?